tsdb/wlog/watcher.go

```diff
@@ -51,6 +51,7 @@ type WriteTo interface {
 	AppendExemplars([]record.RefExemplar) bool
 	AppendHistograms([]record.RefHistogramSample) bool
 	AppendFloatHistograms([]record.RefFloatHistogramSample) bool
+	AppendWALMetadata([]record.RefMetadata) bool
```

Comment:
Can we just name this `AppendMetadata` since we didn't have a way of passing metadata to the WriteTo interface previously?

Reply:
That's a different interface so it should be okay? but I think we should also ensure the scrape code doesn't get a metadata appender if the WAL metadata is being used, ie if both configs are set we should error on startup, or log a warning and only use the metadata

Reply:
> ie if both configs are set we should error on startup, or log a warning and only use the metadata

I also thought to error on startup, but unfortunately the previous mechanism is enabled by default; erroring out directly could be confusing to user which would have to go and explicilty disable the old one.
```
ts=2023-04-28T08:29:04.757Z caller=main.go:1229 level=info msg="Loading configuration file" filename=/Users/tpaschalis/prometheus.yml
ts=2023-04-28T08:29:04.757Z caller=main.go:1255 level=error msg="Failed to apply configuration" err="the 'send_metadata' and 'metadata_config.send' parameters are mutually exclusive"
...
ts=2023-04-28T08:29:04.764Z caller=main.go:1164 level=error err="error loading config from \"/Users/tpaschalis/prometheus.yml\": one or more errors occurred while applying the new configuration (--config.file=\"/Users/tpaschalis/prometheus.yml\")"
```

I think we should go with your other approach; in 42fc28821aafa9f42e6672343e748195ba83aa86, if both are set, I'm logging a warning and defaulting to implcitly disabling the old MetadataWatcher instead. WDYT?

Reply:
As for the method name, there's _another_ `MetadataAppender` interface which clashes with this name.

I've changed _that_ name to AppendMetadataWatcher and kept the simple `AppendMetadata` for our own here. Let me know if you don't want this change and would prefer to roll it back.